File: Corrections/php/Neufplate/src/Avatar.php
<?php

namespace App;

abstract class Avatar
{

    public string $url;

    public function generate(string $hash = null): void {
        $client = $this->getClient();

        if($hash == null) {
            $this->url = $client->getRandomAvatarUrl();
        } else {
            $this->url = $client->getAvatarFromHash($hash);
        }
    }

    protected abstract function getClient(): AvatarClientInterface;

}

File: Corrections/java/Neufplate/src/main/java/fr/neufplate/Avatar.java
package fr.neufplate;

public abstract class Avatar {

    public String url;

    public void generate() {
        AvatarClientInterface client = getClient();
        this.url = client.getRandomAvatarUrl();
    }

    public void generate(String hash) {
        AvatarClientInterface client = getClient();
        this.url = client.getAvatarFromHash(hash);
    }

    protected abstract AvatarClientInterface getClient();

    @Override
    public String toString() {
        return "Avatar{" +
                "url='" + url + '\'' +
                '}';
    }
}